platform/openide.filesystems/src/org/openide/filesystems/FileUtil.java

```diff
@@ -937,6 +938,23 @@ public static FileObject toFileObject(File file) {
     */
     public static void copyAttributes(FileObject source, FileObject dest)
     throws IOException {
+        copyAttributes(source, dest, null);
+    }
+    
+    /** Copies attributes from one file to another.
+    * Note: several special attributes will not be copied, as they should
+    * semantically be transient. These include attributes used by the
```

Comment:
I suggest to define field or function to let people access:
```java
public static final BiFunction<String, Object, Object> DEFAULT_ATTR_TRANSFORMER;
```
then they can either:
* use it as the 3rd parameter to get behavior of the two args method
* delegate to it from own transformer as a last resort
* skip it by not delegating from own transformer

I never really liked the hard-coded list of attributes, but they are useful. Exposing them as `BiFunction` keeps the concept of important attributes on, but puts the full control into the hands of `attrTransformer` writer.

Reply:
Good to mention it now. It requires a semantic change in javadoc which would be hard to do after introduction; thanks.

Reply:
Addressed in ffc97b6